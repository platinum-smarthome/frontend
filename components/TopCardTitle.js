import React, { Component } from 'react'
import { View, Text, Image, StyleSheet } from 'react-native'

export default class TopCardTitle extends Component {
  render() {
    return (
    <View style={ styles.container }>
      <View style={ styles.pad }>
        <Image style={ styles.img } source={ this.props.img } />
      </View>
      <View>
        <Text style={ styles.txt }> Monitor </Text>
      </View>
    </View>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    borderBottomWidth: 1, 
    borderBottomColor: 'rgba(255,255,255, 0.6)', 
    flexDirection: 'row', 
    alignContent: 'flex-start', 
    backgroundColor: '#34b8ed'
  },
  img: {
    width: 25, 
    height: 25 
  },
  pad: {
    paddingHorizontal: 18,
    paddingVertical: 10,
  },
  txt: {
    justifyContent: 'center', 
    color: '#fff', 
    fontSize: 18, 
    fontWeight: '500', 
    paddingVertical: 10
  }
})