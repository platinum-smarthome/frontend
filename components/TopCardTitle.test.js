import React, { Component } from 'react'
import enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import renderer from 'react-test-renderer'

import TopCardTitle from './TopCardTitle'
import { Text, View, Image } from 'react-native'

enzyme.configure({ adapter: new Adapter()})

let wrapper

beforeEach(() => {
  wrapper = shallow(<TopCardTitle />)
})

describe('<TopCardTitle/>', () => {
  it('should render <TopCardTitle/>', () => {
    expect(wrapper).toBeDefined()
  })
})

describe('<TopCardTitle/> child is rendered', () => {
  it('Should render <View/>', () => {
    expect(wrapper.find(<View/>)).toBeDefined()
    expect(wrapper.find('View').length).toBe(3)  
  })
  it('Should render <Text/>', () => {
    expect(wrapper.find(<Text/>)).toBeDefined()
    expect(wrapper.find('Text').length).toBe(1)  
  })
  it('Should render <Image/>', () => {
    expect(wrapper.find(<Image/>)).toBeDefined()
    expect(wrapper.find('Image').length).toBe(1)  
  })
})

describe('snapshot testing', () => {
  it('snapshot', () => {
    const tree = renderer.create(<TopCardTitle/>)
    expect(tree).toMatchSnapshot()
  })
})