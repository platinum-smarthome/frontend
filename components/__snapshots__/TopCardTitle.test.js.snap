// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing snapshot 1`] = `
<View
  style={
    Object {
      "alignContent": "flex-start",
      "backgroundColor": "#34b8ed",
      "borderBottomColor": "rgba(255,255,255, 0.6)",
      "borderBottomWidth": 1,
      "flexDirection": "row",
    }
  }
>
  <View
    style={
      Object {
        "paddingHorizontal": 18,
        "paddingVertical": 10,
      }
    }
  >
    <Image
      source={undefined}
      style={
        Object {
          "height": 25,
          "width": 25,
        }
      }
    />
  </View>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#fff",
          "fontSize": 18,
          "fontWeight": "500",
          "justifyContent": "center",
          "paddingVertical": 10,
        }
      }
    >
       Monitor 
    </Text>
  </View>
</View>
`;
